packages/client/lib/cluster/cluster-slots.ts

```diff
@@ -164,13 +164,14 @@ export default class RedisClusterSlots<
   }
 
   async #discover(rootNode: RedisClusterClientOptions) {
-    this.#resetSlots();
     try {
       const addressesInUse = new Set<string>(),
         promises: Array<Promise<unknown>> = [],
         eagerConnect = this.#options.minimizeConnections !== true;
 
-      for (const { from, to, master, replicas } of await this.#getShards(rootNode)) {
+      const shards = await this.#getShards(rootNode);
+      this.#resetSlots(); // Reset slots AFTER shards have been fetched to prevent a race condition
```

Comment:
This regressed when https://github.com/redis/node-redis/commit/b2d35c528654046eaed5d5d32099de32384400ae landed. 

Is it possible to add a test to prevent further regressions?

Reply:
I dont see an easy way to test it..
I'll merge the PR as is for now, and we can add the test in another PR.. :)